### PR TITLE
Chisel3 related workarounds

### PR DESCRIPTION
**Pull request by palmer-dabbelt**
There's some codegen issues in Chisel 3.  This PR adds some workarounds so we can start building the uncore with Chisel 3 right away, rather than waiting for bug fixes.

### CODE REVIEW

File: src/main/scala/cache.scala [hunk 1]
```diff
@@ -1061,7 +1061,10 @@ class L2WritebackUnit(trackerId: Int)(implicit p: Parameters) extends L2XactTrac
 
   val xact = Reg(new L2WritebackReq)
   val data_buffer = Reg(init=Vec.fill(innerDataBeats)(UInt(0, width = innerDataBits)))
-  val xact_addr_block = Cat(xact.tag, xact.idx, UInt(cacheId, cacheIdBits))
```
Comment (aswaterman) [inline src/main/scala/cache.scala]: Can you make a chisel3 and/or firrtl issue and add a comment here that refers to it? This is certainly "supposed" to work.

**Comment by palmer-dabbelt** [comment]
https://github.com/ucb-bar/chisel3/pull/114

**Comment by hcook** [comment]
Hey, I cherry-picked a5a23b6 and 4175653 into master. I haven't taken 767be08 because it will require bumping rocket-chip/uncore and I'm not quite ready to do that yet. Does the readonly flag on attaching the SCRs really not matter anymore?

**Comment by palmer-dabbelt** [comment]
The read-only flag shouldn't matter: the read-only things we attach to the SCR file are constants, so the right thing will just happen automatically.  I'll deal with this the next time I go and try rocket-chip chisel3, so I'm going to just close this for now.

**Pull request closed without merging**